code/script.js

```diff
@@ -222,104 +230,205 @@ const generateBoard = () => {
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+if (category === 'hair'|| category === 'eyes'){
+  if (keep) {
+  charactersInPlay = charactersInPlay.filter(
+    (person) => person[category] === value
+  )
+  } else {
+  charactersInPlay = charactersInPlay.filter(
+    (person) => person[category] !== value
+  );
+}
+} else if (category === 'accessories' || category === 'other'){
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category].includes(value)
+    );
+  } else {
+    charactersInPlay = charactersInPlay.filter(
+      (person)=> !person[category].includes(value)
+    );
+  }
+}
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+// Invoke this function to redraw the board with the remaining people.
+  generateBoard(); 
+};
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+// when clicking guess, the player first has to confirm that they want to make a guess.
+const guess = (personToConfirm) => {   // store the interaction from the player in a variable.
+  const makeAGuess = confirm(`Are you sure you want to guess on ${personToConfirm}?`)
+  
+  if (makeAGuess) { // If the player wants to guess, invoke the checkMyGuess function.
+    checkMyGuess(personToConfirm)  
+  }
 }
 
+
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+    // 1. Check if the personToCheck is the same as the secret person's name
+  if (personToCheck === secret.name) {
+      // 2. Set a Message to show in the win or lose section accordingly
+      winOrLoseText.innerHTML =  `Hooray, you guessed correctly it was ${secret.name} in ${numberOfGuesses} guesses!`
+      //console.log("yes", personToCheck, secret.name)
+  } else {
+      winOrLoseText.innerHTML = `Sorry, you guessed ${numberOfGuesses} times but it's not ${personToCheck} - it was ${secret.name}! Better luck next time!`
+      //console.log("no", personToCheck, secret.name)
+  }
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex'
   // 4. Hide the game board
+  board.style.display = 'none'
```

Comment:
Another quick FYI - when you hide the board, you'll also need to hide the board wrapper.  Otherwise, you're ending up with a white bar at the bottom of the win and lose screens.  I commented in the HTML which element needs hiding.

Reply:
Hmm I can't see any white bar at the bottom of the win and lose display on any screen size on the Netlify site, but I will have a play with the board wrapper on line 61 and see if anything changes!